lib/src/test/resources/compileContracts.sh

```diff
@@ -1,11 +1,17 @@
-#!/bin/sh
+#!/bin/bash
 
-cd "$(dirname $0)" || exit
+cd "$(dirname "$0")" || exit
+mkdir -p "compiled"
 
-for file in src/*.cairo; do
+# shellcheck disable=SC2044
+for file in $(find src -name "*.cairo" -type f -print); do
```

Comment:
This is prone for whitespace errors, maybe put things from the loop body to separate function and pass it to `-exec` ?